airflow/models/dag.py

```diff
@@ -1966,7 +1966,7 @@ def deactivate_stale_dags(expiration_date, session=None):
             )
             dag.is_active = False
             session.merge(dag)
-            session.commit()
+        session.commit()
```

Comment:
Similar to `deactivate_unknown_dags()`, commit only once outside the for-loop.

Reply:
This should actually just be
```suggestion
        session.flush()
```

to be in line with https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#database-session-handling

Reply:
In this case, only in `model/dag.py" there are also a few other usage of `session.commit()` when `@provide_session` is used. Address them in this PR as well?

Reply:
Possibly -- do you think they make sense all as one PR?

Reply:
How about this: I will skip it in this PR, and have another PR dedicated for clearing `session.commit()` project-wise. So the PR scopes are clearer. Agree?

Reply:
I created issue https://github.com/apache/airflow/issues/12818 for clearing `session.commit()`.

Given it's a relatively easy fix to do, I mark it as "**good first issue**" and let's see if any new-contributor would like to pick it up (will voice in Slack).